### PR TITLE
#80 - Inviting to quizzes - Backend

### LINKED ISSUE #80: Inviting to quizzes - Backend
Implement the backend for the inviting page.

- [x] Option to invite group of user to test
- [x] Sorting records by school, score, or username
- [x] Option to group records by school, quiz
- [x] Option to limit the displayed records per group to a configurable number
- [x] Pagination

Example search query
Here are some examples of the expected search queries that the backend should support:
```?groupBy=[school,quiz]&limitBy=username&limit=4&desc```: Group the results by quiz and then by school, limit by username, show 4 records per group, and sort in descending order.
```?groupBy=[school]&limitBy=score&limit=4&desc&quiz=5```: Group by school, limit by score, show 4 records per group, sort in descending order, and filter by quiz with the ID 5.

### PR DESCRIPTION
**Pull request by PrabuckiDominik**
This should close #80 .

### CODE REVIEW

**Comment by PrabuckiDominik** [comment]
Need to add tests

File: app/Http/Resources/InviteResource.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
```
Comment (AmonDeShir) [inline app/Http/Resources/InviteResource.php:1]: It seems that `InviteResource` is not being used anywhere. Consider removing it.

File: resources/js/Types/Invite.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {type School} from '@/Types/School'
```
[the remark this answers is not in the document]
Comment (AmonDeShir) [reply resources/js/Types/Invite.d.ts:1]: This file looks almost identical to user.d.ts. It seems redundant.

**Review by AmonDeShir — CHANGES_REQUESTED** [review]

File: resources/js/Pages/Admin/Invite.vue [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+const toggleUserSelection = (userId: number) => {
+  if (selectedUserIds.value.includes(userId)) {
+    selectedUserIds.value = selectedUserIds.value.filter(id => id !== userId)
+  } else {
+    selectedUserIds.value.push(userId)
+  }
+}
+
+const showPagination = computed(() => {
+  return params.limit === null
```
Comment (JokeUrSelf) [inline resources/js/Pages/Admin/Invite.vue:64]: In `useParams` you had `undefined` return type, but here you compare one of the parameters to `null`. `null` != `undefined`. To make this code work for `null`, `undefined` and empty string, you can:
 ```suggestion
   return !params.limit
```

**Review by AmonDeShir — DISMISSED** [review]

File: resources/views/emails/auth/invite-user.blade.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+<header>
+    <img src="" alt="{{ config('app.name') }} Logo" width="150" height="50">
+    <h1>{{ config('app.name') }}</h1>
+</header>
+
+<body>
+<h1>Cześć {{ $user->name }},</h1>
+
+<p>Zostałeś zaproszony do udziału w quizie: {{ $quiz->name }}.</p>
```
Comment (EwelinaSkrzypacz) [inline resources/views/emails/auth/invite-user.blade.php:18]: How about girls? "Zostałeś/aś"? Or maybe "Zapraszamy Cię do udziału..."

Comment (AmonDeShir) [reply resources/views/emails/auth/invite-user.blade.php]: Fixed

**Review by AmonDeShir — APPROVED** [review]

**Pull request merged**